Doc/library/unittest.mock.rst

```diff
@@ -1331,6 +1332,8 @@ magic methods :meth:`__getitem__`, :meth:`__setitem__`, :meth:`__delitem__` and
     >>> assert thing['one'] == 1
     >>> assert list(thing) == ['one']
 
+:func:`patch.dict` returns a patched dictionary when used as context manager.
```

Comment:
This needs to be moved after the paragraph that starts with ":func:\`patch.dict\` can be used as a context manager, [...]".

Also, add a versionchanged marker after that paragraph:

```rst
.. versionchanged:: 3.7
   :func:`patch.dict` now returns the patched dictionary when it used as a context manager.
```

(Feel free to use any wording you want, I just gave an example.)